src/main/scala/diplomacy/Nodes.scala

```diff
@@ -5,123 +5,328 @@ package freechips.rocketchip.diplomacy
 import Chisel._
 import chisel3.experimental.IO
 import chisel3.internal.sourceinfo.SourceInfo
-import freechips.rocketchip.config.{Parameters,Field}
+import freechips.rocketchip.config.{Field, Parameters}
 import freechips.rocketchip.util.HeterogeneousBag
+
+import scala.collection.immutable
 import scala.collection.mutable.ListBuffer
-import scala.util.matching._
 
+/** A field available in [[Parameters]] used to determine whether [[InwardNodeImp.monitor]] will be called.
+ */
 case object MonitorsEnabled extends Field[Boolean](true)
+
+/** When rendering the edge in graphical format (e.g. GraphML), flip the direction. */
```

Comment:
This comment is not quite correct. The arrow for the edge stays in the same direction.

Normally, yEd tries to put the source node vertically above the sink node. RenderFlipped inverts this. In other words, the sink gets place vertically above the source. This is useful for things like Tiles which include both masters and slaves. You want the tiles at the top of the diagram. The master interfaces for the tile will be correctly placed vertically above the rest of the SoC. However, the slave interfaces would normally get placed near the bottom of the SoC. RenderFlipped on the edges leading to control tile slaves ensures that the tile can be placed at the top of the diagram despite this.